rfcs/react-components/convergence/slot-children-render-function.md

```diff
@@ -0,0 +1,650 @@
+
+##### Cons
+
+1. Requires extra build steps
+2. Requires a small migration (but it can be done granularly, nothing will break)
+
+### Option B: Refactor getSlots + helper method
+
+An alternative solution would be to stop children function rendering on `getSlots` method (since we don't have enough information at that moment)
+and then wrapping all children overrides with a helper method (let's call it `resolveChildren`):
+
+```tsx
+// renderAccordionHeader
+<slots.root {...slotProps.root}>
+  {resolveChildren(
+    slotProps.root,
+    <slots.button {...slotProps.button}>
+      {resolveChildren(
+        slotProps.button,
+        <>
+          {state.expandIconPosition === 'start' && (
+            <slots.expandIcon {...slotProps.expandIcon}>
+              {resolveChildren(
+                slotProps.expandIcon,
+                <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
+              )}
+            </slots.expandIcon>
+          )}
+          {slots.icon && <slots.icon {...slotProps.icon} />}
+          {slotProps.root.children}
+          {state.expandIconPosition === 'end' && (
+            <slots.expandIcon {...slotProps.expandIcon}>
+              {resolveChildren(
+                slotProps.expandIcon,
+                <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
+              )}
+            </slots.expandIcon>
+          )}
+        </>,
+      )}
+    </slots.button>,
+  )}
+</slots.root>
+```
+
+#### Pros and Cons
+
+##### Pros
+
+1. It would solve the problem?!
+
+##### Cons
+
+1. Render methods will become more complex
+
+### Option C: Custom JSX Pragma without internal changes
+
+A minimal version of option A, but without introducing internal changes. If we focus on the problem itself,
+slot children render function, the main requirement would be to maintain `defaultProps.children` until the last possible
+moment so that it can be introduced once again back to a children render function.
+
+This proposal would take advantage of `Symbol` to introduce a "private" method to the result of `resolveShorthand`,
+to maintain the value of `defaultProps.children`. That value would then later on be consumed by a minimal custom JSX pragma
+that on the specific case of children render function this pragma would consume this `defaultProps.children` to ensure proper functionality.
+
+```diff
+const defaultPropsChildrenSymbol = Symbol('fuiSlotDefaultPropsChildren')
+
+export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
+  //...
+
+  return defaultProps ? {
+    ...defaultProps,
+    ...resolvedShorthand,
++   [defaultPropsChildrenSymbol]: defaultProps?.children
+  } : resolvedShorthand;
+};
+```
+
+https://codesandbox.io/s/wispy-leftpad-f8yi37?file=/example.tsx.
+
+#### Pros and Cons
+
+##### Pros
+
+1. No internal changes required
+2. No migration is required
+
+##### Cons
+
+1. Requires extra build step
```

Comment:
I don't think so, we just need to put a magic comment to `render*.tsx` files i.e.

```diff
+/** @jsxRuntime classic */
+/** @jsx fluentCreateElement */

```

Reply:
yeaaah, I was trying to hide this, as it honestly sounds like a hack to add comments to every single file we want to change pragma. I'd rather spend the time to update our build process to simply include the pragma change, sounds more elegant.